Add `Segmented Group` documentation (#1282)
hashicorp/design-system: website/docs/components/form/segmented-group/partials/guidelines/guidelines.md

## Usage

> A Segmented Group is an agnostic container that wraps two or more `Segments`, making the relevant use cases broad.

Reviewer: No need of the "code" visual treatment: that's not the name of a component.

```suggestion
A Segmented Group is an agnostic container that wraps two or more Segments, making the relevant use cases broad.
```

!!! Info

> The way in which these components are published and assembled is fundamentally different between the Figma and Ember components.

Reviewer 2: ```suggestion
How these components are published and assembled is fundamentally different between the Figma and Ember components.
```

- In Figma, we publish multiple primitive `Segment` components (`Segmented Button`, `Segmented Toggle`, and `Segmented Input`) that are intended to be assembled in an auto layout container. A pre-assembled `Base` component is also made available for more simple instances.

> - The Ember component is a container that enforces consistent styling and grouping of nested Helios form components. It does not manage any logic or functionality out of the box, these are left up to the consumer to implement based on the nested components.

Reviewer: ```suggestion
- The Ember component is a container that enforces consistent styling and grouping of nested Helios input fields and action components. It does not manage any logic or functionality out of the box, these are left up to the consumer to implement based on the nested components.
```

Author: Ah good catch there 👏

!!!

### When to use

- When filtering complex data, searching within a set of filters, and in creating complex filter patterns.
- When collecting information in a way that benefits from a "connected" visual representation, e.g., a key/value pair.
- When an action performs a generative function rather than a submit function, e.g., generating API credentials.
- As a complex data input within a form.

### When not to use

> - To submit a form or set of inputs. A Segmented Group can be used within a form to collect complex data, but should not submit that data via a form method. Use the guidelines in [Form patterns](/patterns/form-patterns) instead.

Reviewer 2: 🤔 hmm, not sure about this – why would we not want them to submit the data using a standard form? happy to catch-up on this

Reviewer 2: ah, I see! yes, that sounds like valuable guidance – may not require rewording, it was probably just me reading through the lines 😅

- To connect multiple fields that each require their own label.
- When adding a prefix or suffix to an input field. This is not currently supported by Helios form components, instead consider using a simple text element preceding or following the form element.

## Segment components

> A Segmented Group can consist of multiple primitive Segments that vary in their properties and intended input value. We recommend limiting usage in Figma to the defined `Segmented Button`, `Segmented Toggle` and `Segmented Input` primitives to ensure consistent styling and grouping. The Ember component supports a generic element in either the leading or trailing position if a custom solution is required.

Reviewer: See comment below about "Toggle" vs "Dropdown" (and check in other parts of the documentation too)

```suggestion
A Segmented Group can consist of multiple primitive Segments that vary in their properties and intended input value. We recommend limiting usage in Figma to the defined `Segmented Button`, `Segmented Dropdown` and `Segmented Input` primitives to ensure consistent styling and grouping. The Ember component supports a generic element in either the leading or trailing position if a custom solution is required.
```

Reviewer: @alex-ju the statement "The Ember component supports a generic element in either the leading or trailing position if a custom solution is required" I think is not accurate: since it's a yielded component, it can be in any position.

Reviewer 2: Indeed, it can be anywhere, leading, trailing, or anywhere in the middle.

### Segmented Button

A `Segmented Button` can perform a function or interaction within a Segmented Group, e.g., generating an object, string, or value in a field it is connected to. It is an extension of the Helios [Button](/components/button).

> ### Segmented Toggle

Reviewer: [blocker] What it's called here "Toggle" is technically a "Dropdown", we should use the correct term otherwise it can be confusing (a form Toggle is [something different](https://helios.hashicorp.design/components/form/toggle))

(check also in other parts of the documentation, and Figma too I think)

> A `Segmented Toggle` can be used to filter or narrow the available parameters when searching within a dataset or group of objects. This is an extension of the Helios [Toggle Button](/components/dropdown#toggle) and is often used in the context of a [Dropdown](/components/dropdown).

Reviewer: Unless I am missing something, this is not factually correct: it's a Dropdown, not just an extension of the Dropdown toggle. Right?
